ui/src/app/components/runs/runs.component.ts

```diff
@@ -62,18 +69,52 @@ export class RunsComponent implements OnDestroy, AfterViewInit {
   }
 
   onClarityDgRefresh(state: ClrDatagridStateInterface) {
-    let sort: SortModel = state.sort ? new SortModel(<string>state.sort.by, state.sort.reverse ? -1 : 1) : undefined;
+    this.sort = state.sort ? new SortModel(<string>state.sort.by, state.sort.reverse ? -1 : 1) : undefined;
+    this.pageFrom = state.page.from < 0 ? 0 : state.page.from;
+    this.pageSize = state.page.size;
+
+    let filters:{[prop:string]: any} = {};
+    if (state.filters) {
+      for (let filter of state.filters) {
+        let {property, value} = <{property: string, value: any}>filter;
+        filters[property] = value;
+      }
+    }
+    this.filters = filters;
+
+    this.refresh();
```

Comment:
You can pass filters directly into `this.refresh()`. Then you don't need `this.filters` at all.

Reply:
Yes, I do need, because I call refresh method in view, where I don't have these values.

Reply:
Ok I see